idunn/datasources/hove/models.py

```diff
@@ -0,0 +1,338 @@
+                    properties={"mode": self.mode.value if self.mode else "tram"},
+                    mode=mode,
+                )
+                for (inst, geo) in insts
+            ]
+
+        return api.RouteLeg(
+            duration=self.duration,
+            distance=self.geojson["properties"][0]["length"],
+            summary="todo",
+            steps=steps,
+            info=(
+                self.display_informations.as_api_transport_info()
+                if self.display_informations
+                else None
+            ),
+            **(
+                {
+                    "from": stops[0],
+                    "stops": stops[1:-1],
+                    "to": stops[-1],
+                }
+                if stops
+                else {}
+            ),
+        )
+
+
+class Journey(BaseModel):
+    arrival_date_time: datetime
+    departure_date_time: datetime
+    duration: int
+    distances: Distances
+    sections: List[Section]
+
+    @validator("arrival_date_time", "departure_date_time", pre=True)
+    def parse_date(cls, v):
+        return datetime.strptime(v, "%Y%m%dT%H%M%S")
+
+    def as_api_route(self) -> api.DirectionsRoute:
+        secs = [sec for sec in self.sections if sec.sec_type != SectionType.WAITING]
+        legs = [sec.as_api_route_leg() for sec in secs]
+
+        summary = [
+            summary
+            for sec in self.sections
+            if (summary := sec.as_api_route_summary_part()).info is not None
+        ]
+
+        geometry = {
+            "type": "FeatureCollection",
+            "features": [
+                {
+                    "type": "Feature",
+                    "properties": {
+                        "mode": leg.mode.value,
+                        **(
+                            {
+                                "lineColor": sec.display_informations.color,
+                                "lineTextColor": sec.display_informations.text_color,
+                            }
+                            if sec.display_informations
+                            else {}
+                        ),
+                    },
+                    "geometry": {
+                        "type": "MultiLineString",
+                        "coordinates": [step.geometry["coordinates"] for step in leg.steps],
+                    },
+                }
+                for sec, leg in zip(secs, legs)
+                for step in leg.steps
+            ],
+        }
+
+        return api.DirectionsRoute(
+            duration=self.duration,
+            distance=self.distances.overall(),
+            start_time=datetime.isoformat(self.departure_date_time),
+            end_time=datetime.isoformat(self.arrival_date_time),
+            legs=legs,
+            geometry=geometry,
+            summary=summary,
+        )
+
+
+class HoveResponse(BaseModel):
+    journeys: List[Journey] = []
+
+    def as_api_response(self) -> api.DirectionsResponse:
```

Comment:
why is there a `as` in this method name and in the other ones? Is it to specify that this method is a conversion to the qwant direction model response ?

Reply:
Yeah, this is the idea I wanted to convey with the `as_api_` prefix.

As you are asking this I suppose it doesn't do a great job of conveying this idea ? :cry: